pkg/resource/db_instance/hooks.go

```diff
@@ -160,3 +177,826 @@ func instanceDeleting(r *resource) bool {
+		f66 := []*svcapitypes.DBInstanceStatusInfo{}
+		for _, f66iter := range resp.DBInstance.StatusInfos {
+			f66elem := &svcapitypes.DBInstanceStatusInfo{}
+			if f66iter.Message != nil {
+				f66elem.Message = f66iter.Message
+			}
+			if f66iter.Normal != nil {
+				f66elem.Normal = f66iter.Normal
+			}
+			if f66iter.Status != nil {
+				f66elem.Status = f66iter.Status
+			}
+			if f66iter.StatusType != nil {
+				f66elem.StatusType = f66iter.StatusType
+			}
+			f66 = append(f66, f66elem)
+		}
+		ko.Status.StatusInfos = f66
+	} else {
+		ko.Status.StatusInfos = nil
+	}
+	if resp.DBInstance.StorageEncrypted != nil {
+		ko.Spec.StorageEncrypted = resp.DBInstance.StorageEncrypted
+	} else {
+		ko.Spec.StorageEncrypted = nil
+	}
+	if resp.DBInstance.StorageType != nil {
+		ko.Spec.StorageType = resp.DBInstance.StorageType
+	} else {
+		ko.Spec.StorageType = nil
+	}
+	if resp.DBInstance.TagList != nil {
+		f69 := []*svcapitypes.Tag{}
+		for _, f69iter := range resp.DBInstance.TagList {
+			f69elem := &svcapitypes.Tag{}
+			if f69iter.Key != nil {
+				f69elem.Key = f69iter.Key
+			}
+			if f69iter.Value != nil {
+				f69elem.Value = f69iter.Value
+			}
+			f69 = append(f69, f69elem)
+		}
+		ko.Status.TagList = f69
+	} else {
+		ko.Status.TagList = nil
+	}
+	if resp.DBInstance.TdeCredentialArn != nil {
+		ko.Spec.TDECredentialARN = resp.DBInstance.TdeCredentialArn
+	} else {
+		ko.Spec.TDECredentialARN = nil
+	}
+	if resp.DBInstance.Timezone != nil {
+		ko.Spec.Timezone = resp.DBInstance.Timezone
+	} else {
+		ko.Spec.Timezone = nil
+	}
+	if resp.DBInstance.VpcSecurityGroups != nil {
+		f72 := []*svcapitypes.VPCSecurityGroupMembership{}
+		for _, f72iter := range resp.DBInstance.VpcSecurityGroups {
+			f72elem := &svcapitypes.VPCSecurityGroupMembership{}
+			if f72iter.Status != nil {
+				f72elem.Status = f72iter.Status
+			}
+			if f72iter.VpcSecurityGroupId != nil {
+				f72elem.VPCSecurityGroupID = f72iter.VpcSecurityGroupId
+			}
+			f72 = append(f72, f72elem)
+		}
+		ko.Status.VPCSecurityGroups = f72
+	} else {
+		ko.Status.VPCSecurityGroups = nil
+	}
+
+	rm.setStatusDefaults(ko)
+
+	// We expect the DB instance to be in 'creating' status since we just
+	// issued the call to create it, but I suppose it doesn't hurt to check
+	// here.
+	if instanceCreating(&resource{ko}) {
+		// Setting resource synced condition to false will trigger a requeue of
+		// the resource. No need to return a requeue error here.
+		ackcondition.SetSynced(&resource{ko}, corev1.ConditionFalse, nil, nil)
+		return &resource{ko}, nil
+	}
+	return &resource{ko}, nil
+}
+
+// restoreDbInstanceFromDbSnapshotPayload returns an SDK-specific struct for the HTTP request
+// payload of the RestoreDbInstanceFromDbSnapshot API call
+func (rm *resourceManager) restoreDbInstanceFromDbSnapshotPayload(
+	r *resource,
+) (*svcsdk.RestoreDBInstanceFromDBSnapshotInput, error) {
+	res := &svcsdk.RestoreDBInstanceFromDBSnapshotInput{}
+
+	if r.ko.Spec.DBSnapshotIdentifier != nil {
```

Comment:
Likewise we should be able to auto-generate this

Reply:
synced offline, addressed in next revision